src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2201,7 +2188,9 @@ public FilePreferences getFilePreferences() {
 
         filePreferences = new FilePreferences(
                 getInternalPreferences().getUser(),
-                determineMainFileDirectory(get(MAIN_FILE_DIRECTORY)),
+                StringUtil.isNotBlank(get(MAIN_FILE_DIRECTORY))
```

Comment:
Refactor to 
```suggestion
getPath(MAIN_FILE_DIRECTORY).orElse(JabRefDesktop.getNativeDesktop().getDefaultFileChooserDirectory())                
```

Reply:
There's so such nethod. And I don't see the benefit of wrapping it an optional here.

Reply:
StringUtil.isNotBlank(...) is - I believe - the equivalent to `... != null && !....isEmpty()", so should simplify determineMainFileDirectory

Reply:
On the other hand, method get is called twice. Maybe we can extract the var from the constructor call

Reply:
Since we store quite a few paths in the preferences, it makes sense to introduce a helper for it in my opinion. You can also introduce a second default arg:
```java
getPath(string name, Path defaultVal) {
   val = get(name)
   return StringUtil.isNotBlank(val) ? Path.of(val) : defaultVal
}
```

Reply:
I've reintroduced a method